Replace npm with yarn in exo-lint

martinjaime opened the pull request:
<!-- mention the issue this PR addresses -->
Addresses Originate/exosphere-sdk#170

<!-- a short description of the change in addition to what is already mentioned in the issue above -->

<!-- tag a few reviewers -->

Commit:
Replaces npm with yarn in exo-lint

Commit:
Adds badge to README

Commit:
Check if yarn is installed

and output a useful error message if not

Commit:
add yarn.lock file

Commit:
Accidentally added wrong lock file on previous commit

fixed

martinjaime commented:
@kevgo, seems like tests are failing due to docker issues.

kevgo commented on exo-lint/bin/setup:
```diff
@@ -1,2 +1,13 @@
-npm install --loglevel error --depth 0
+
+# check if yarn is installed
+command -V yarn >/dev/null 2>&1
```
Is `command` available on all platforms? What about using `which yarn` here?

Commit:
Merge branch 'master' into mj-yarn-exo-lint

kevgo commented on exo-lint/bin/setup:
```diff
@@ -1,2 +1,13 @@
+# check if yarn is installed
+command -V yarn >/dev/null 2>&1
+
+if [ $? -eq 0 ]; then
```
Check for the error first and exit, then do the actual thing later.

```bash
if [ $? -gt 0 ]; then
  echo "ERROR...
  exit 1
fi

yarn install

Commit:
Make more compatible

 - use `which` command instead of `command`
 - simplifies error handler

Commit:
Workaround bug in dependency-lint

it is searching the entire dependency tree for exeucutables when it
should only search the ones that are exposed.
